src/endpoint/s3/s3_rest.js

```diff
@@ -139,6 +139,8 @@ async function handle_request(req, res) {
     authenticate_request(req);
     await authorize_request(req);
 
+    await http_utils.set_expiration_header(req, res);
```

Comment:
Why set the header `rest_s3`?
It seems like you're adding the header for all 'HEAD' 'GET' 'PUT' methods while it is expected in the `HeadObject`, `GetObject`, and `PutObject`, right?
I would move it to the specific relevant ops and avoid a change here.